glib-macros/src/object_interface_attribute.rs

```diff
@@ -22,16 +22,15 @@ pub fn impl_object_interface(input: &mut syn::ItemImpl) -> TokenStream {
         ..
     } = input;
 
-    let self_ty_as_ident = match &**self_ty {
```

Comment:
I didn't look too close but I think this is currently unsound. There's only a single type registered for every `T`, so you can currently safely cast between different `T`. Not sure how that can be solved nicely.

Reply:
Unlike in Java this is unsound here though :) In the simplest case, you could have a `MyObj<T>` that stores a `T`, and then simply casting that between `T: String` to `T: u32` and the subclass couldn't know about that happening and accesses the value with the wrong type then.